Ghidra/Processors/V850/data/languages/Instructions/Arithmetic.sinc

```diff
@@ -0,0 +1,336 @@
+{
+	R1115 = R0004 + s1631;
+}
+
+# MOVHI imm16, reg1, reg2 - rrrrr110010RRRRR|iiiiiiiiiiiiiiii
+:movhi s1631, R0004, R1115 is op0510=0x32 & op1115!=0 & R0004 & R1115; s1631
+{
+	R1115 = R0004 + (s1631 << 16);
+}
+
+# SUB reg1, reg2 - rrrrr001101RRRRR
+:sub R0004, R1115 is op0510=0x0D & R0004 & R1115 
+{
+	set_general_flags_neg(R1115, R0004);
+	R1115 = R1115 - R0004;
+}
+
+# SUBR reg1, reg2 - rrrrr001100RRRRR
+:subr R0004, R1115 is op0510=0x0C & R0004 & R1115
+{
+	set_general_flags_neg(R0004, R1115);
+	R1115 = R0004 - R1115;
+}
+
+
+
+#####################################################
+#####		       Saturated		#####
+#####################################################
+
+
+define pcodeop __saturate;
+
+# SATADD reg1, reg2 - rrrrr000110RRRRR
+:satadd R0004, R1115 is op0510=0x06 & R0004 & R1115 & op1115!=0
+{
+	set_general_flags_pos(R1115, R0004);
+	$(SAT) = $(SAT) || $(OV);
+	R1115 = R1115 + R0004;
+	__saturate(R1115);
+}
+
+#SATADD imm5, reg2 - rrrrr010001iiiii
+:satadd s0004, R1115 is op0510=0x11 & s0004 & R1115 & op1115!=0
+{
+	set_general_flags_pos(R1115, s0004);
+	$(SAT) = $(SAT) || $(OV);
+	R1115 = R1115 + s0004;
+	__saturate(R1115);
+}
+
+# SATADD reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww01110111010
+:satadd R0004, R1115, R2731 is op0510=0x3F & R0004 & R1115; op1626=0x3BA & R2731
+{
+	set_general_flags_pos(R1115, R0004);
+	$(SAT) = $(SAT) || $(OV);
+	R2731 = R1115 + R0004;
+	__saturate(R2731);
+}
+
+# SATSUB reg1, reg2 - rrrrr000101RRRRR
+:satsub R0004, R1115 is op0510=0x05 & R0004 & R1115 & op1115!=0 
+{
+	set_general_flags_neg(R1115, R0004);
+	$(SAT) = $(SAT) || $(OV);
+	R1115 = R1115 - R0004;
+	__saturate(R1115);
+}
+
+# SATSUB reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww01110011010
+:satsub R0004, R1115, R2731 is op0510=0x3F & R0004 & R1115; op1626=0x39A & R2731
+{
+	set_general_flags_neg(R1115, R0004);
+	$(SAT) = $(SAT) || $(OV);
+	R2731 = R1115 - R0004;
+	__saturate(R2731);
+}
+
+# SATSUBI imm16, reg1, reg2
+:satsubi s1631, R0004, R1115 is op0510=0x33 & op1115!=0 & R0004 & R1115; s1631
+{
+	set_general_flags_neg(R0004, s1631);
+	$(SAT) = $(SAT) || $(OV);
+	R1115 = R0004 - s1631;
+	__saturate(R1115);
+}
+
+# SATSUBR reg1, reg2
+:satsubr R0004, R1115 is op0510=0x04 & R0004 & R1115 & op1115!=0
+{
+	set_general_flags_neg(R0004, R1115);
+	$(SAT) = $(SAT) || $(OV);
+	R1115 = R0004 - R1115;
+	__saturate(R1115);
+}
+
+
+
+#####################################################
+#####                  Divide			#####
+#####################################################
+
+
+# DIV reg1, reg2, reg3 - rrrrr111111RRRRR|wwwww01011000000
+:div R0004, R1115, R2731 is op0510=0x3F & R0004 & R1115; op1626=0x2C0 & R2731
+{
+	local quot:4 = R1115 s/ R0004;
+	local mod:4 = R1115 s% R0004;
+	$(OV) = ((R1115 == 0x80000000 && R0004 == 0xFFFFFFFF) || R0004 == 0x0);
+	set_Z(R1115);
+	set_S(R1115);
+	either_or(R1115, (R1115 == R2731), mod, quot);	#if reg2 == reg3 the remainder is stored, if not the quotient is stored
```

Comment:
I don't think you need to take this either_or step. Just assign R1115 = quot here. If R1115 = R2731, it will be correctly assigned mod in the following step. Ditto for the other div operations.